### DISCUSSION TITLE
How to convert OR-decision table to decision tree(s) or DAG

### DISCUSSION THREAD

**Discussion by jet10000** [kayjan/bigtree#114] [Questions]
How to do it?

OR-decision table like this:

![image](https://github.com/kayjan/bigtree/assets/2258120/13fcc286-32f1-4f31-9787-795e6d625313)

Decision tree like this:

![image](https://github.com/kayjan/bigtree/assets/2258120/3f645570-9e07-4a28-9e06-61f52d2489da)

**Comment by kayjan** [comment]
Hello, I have a few clarifications

1. How do I read the OR table?
a) Does it mean that if there is evidence E01, it represents the presence of disease D01?
b) Why are there duplicates (i.e., D01, D03, D04) in the Disease column?
c) Does Y represents 1 and T represent 0 when it is represented on the decision tree?
2. How do I link the OR table to the decision tree?
a) The OR table seems to be two dimensional with evidence and disease, how do you expect it to be reflected on the decision tree?
b) Is there a smaller example with actual values? I would think that the OR table should be a pandas DataFrame or a nested list, would be good to have some skeleton code of the expected input of the OR table 👍

**Comment by jet10000** [comment]
How do I read the OR table?

a) Does it mean that if there is evidence E01, it represents the presence of disease D01?

Yes

b) Why are there duplicates (i.e., D01, D03, D04) in the Disease column?

Because disease is only the result, and the cause of the disease is multifactorial.

c) Does Y represents 1 and T represent 0 when it is represented on the decision tree?

YES

How do I link the OR table to the decision tree?

a) The OR table seems to be two dimensional with evidence and disease, how do you expect it to be reflected on the decision tree?

One tree or more tree 

<img width="507" alt="image" src="https://github.com/kayjan/bigtree/assets/2258120/1d0df7ed-b5fb-4cef-ac43-02935991f46d">

or

<img width="397" alt="image" src="https://github.com/kayjan/bigtree/assets/2258120/ec52a4ce-a809-466f-99de-e6f9eca96490">


b) Is there a smaller example with actual values? I would think that the OR table should be a pandas DataFrame or a nested list, would be good to have some skeleton code of the expected input of the OR table 👍

https://docs.google.com/spreadsheets/d/1x3KijUdAONo0dd7j0Hrm89jJadVJMnhlHAbIicnxA08/edit?usp=sharing

[examaple.csv](https://github.com/kayjan/bigtree/files/13261754/examaple.csv)

**Comment by kayjan** [comment]
Sorry I'm still confused at this usage.

1a) If it is multifactorial (from your answer in 1b), does it mean evidence E01 OR E02 OR E03 signifies the presence of disease D01? Then what is the purpose of having a duplicated disease row if any evidence signifies the presence of disease?

1b) If it is multifactorial, in the first row, E01 OR E02 OR E03 signifies the presence of disease D01. In the second row, E01 OR E02 signifies the presence of disease D01 as well, then is the first row redundant? **Perhaps more context on the disease being multifactorial, the link between evidence and disease, and how to interpret the OR decision table would be great**.

2a) Are the tree diagrams referring to your example OR table above? I'm confused why does the parent-child relationship of `e01 -> e02 -> e03 -> d01` (first diagram) exist. Are the evidence dependent on each other? Is this evidence dependency captured in the OR decision table?

And in the parent-child relationship of `e05 -> e04 -> e03 -> d01` (first diagram), why is e05 and e04 related to disease d01? Is this information captured in the OR decision table?

You mentioned the OR table can be represented by one or more trees, how would multiple trees be created from one OR table?

2b) **A small working example of the OR decision table (in CSV) and the final expected tree diagram (in PNG) would be great and help me interpret what you are trying to achieve here**. The attached CSV is too huge for an initial example.

**Comment by kayjan** [comment]
Closing this issue due to insufficient description and inactivity
